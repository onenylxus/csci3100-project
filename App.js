// Import
import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import ForgotPasswordStack from './screens/ForgotPasswordStack';
import Header from './components/Header';
import HomeTab from './screens/navigators/HomeTab';
import LoginScreen from './screens/LoginScreen';
import RegisterStack from './screens/navigators/RegisterStack';
import VerificationScreen from './screens/VerificationScreen';

// Stack
const Stack = createStackNavigator();

// Export application
export default function App() {
  function NoHeader() {
    return { headerShown: false };
  }
  // prettier-ignore
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Login">
        <Stack.Screen name="ForgotPassword" component={ForgotPasswordStack} options={NoHeader} />
        <Stack.Screen name="Tabs" component={HomeTab} options={Header} />
        <Stack.Screen name="Verification" component={VerificationScreen} options={NoHeader} />
        <Stack.Screen name="Login" component={LoginScreen} options={Header} />
        <Stack.Screen name="Register" component={RegisterStack} options={Header} />
      </Stack.Navigator>
      
      <StatusBar style="auto" />
    </NavigationContainer>
  );
}
