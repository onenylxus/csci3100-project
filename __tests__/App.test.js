// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import App from '../App';

// Mock FontAwesome icons
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

// Run jest tests
describe('App', () => {
  let element;

  beforeEach(() => {
    element = render(<App />);
  })

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
