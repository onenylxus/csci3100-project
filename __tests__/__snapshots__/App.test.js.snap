// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App compiles successfully 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#69C6F0",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                CU There
              </Text>
            </View>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "flex-end",
                  "bottom": 0,
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <Image
                  capInsets={true}
                  source={1}
                  style={
                    Object {
                      "borderRadius": 16,
                      "height": 32,
                      "margin": 8,
                      "width": 32,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#ffffff",
                          "flex": 1,
                          "justifyContent": "center",
                          "paddingHorizontal": 32,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "fontSize": 20,
                            "marginBottom": 20,
                          }
                        }
                      >
                        CU There
                      </Text>
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "flexDirection": "column",
                              "justifyContent": "center",
                              "marginBottom": 16,
                              "width": "95%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "flex-start",
                                "marginLeft": "10%",
                              }
                            }
                          >
                            Username:
                          </Text>
                          <View
                            style={
                              Object {
                                "alignSelf": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#000",
                                "borderRadius": 5,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "height": 40,
                                "marginVertical": "5%",
                                "paddingHorizontal": "3%",
                                "width": "85%",
                              }
                            }
                          >
                            <
                              icon={
                                Object {
                                  "icon": Array [
                                    448,
                                    512,
                                    Array [],
                                    "f007",
                                    "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
                                  ],
                                  "iconName": "user",
                                  "prefix": "fas",
                                }
                              }
                              size={15}
                              style={
                                Object {
                                  "marginHorizontal": 5,
                                  "marginVertical": 12,
                                }
                              }
                            />
                            <TextInput
                              allowFontScaling={true}
                              onChangeText={[Function]}
                              placeholder="Username"
                              rejectResponderTermination={true}
                              style={
                                Object {
                                  "width": 190,
                                }
                              }
                              underlineColorAndroid="transparent"
                            />
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "flexDirection": "column",
                              "justifyContent": "center",
                              "marginBottom": 16,
                              "width": "95%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "flex-start",
                                "marginLeft": "10%",
                              }
                            }
                          >
                            Password:
                          </Text>
                          <View
                            style={
                              Object {
                                "alignSelf": "center",
                                "backgroundColor": "#fff",
                                "borderColor": "#000",
                                "borderRadius": 5,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "height": 40,
                                "marginVertical": "5%",
                                "paddingHorizontal": "3%",
                                "width": "85%",
                              }
                            }
                          >
                            <
                              icon={
                                Object {
                                  "icon": Array [
                                    448,
                                    512,
                                    Array [],
                                    "f023",
                                    "M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z",
                                  ],
                                  "iconName": "lock",
                                  "prefix": "fas",
                                }
                              }
                              size={15}
                              style={
                                Object {
                                  "marginHorizontal": 5,
                                  "marginVertical": 12,
                                }
                              }
                            />
                            <TextInput
                              allowFontScaling={true}
                              clearTextOnFocus={false}
                              enablesReturnKeyAutomatically={true}
                              onChangeText={[Function]}
                              placeholder="Password"
                              rejectResponderTermination={true}
                              secureTextEntry={true}
                              style={
                                Object {
                                  "width": 165,
                                }
                              }
                              underlineColorAndroid="transparent"
                            />
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <
                                icon={
                                  Object {
                                    "icon": Array [
                                      640,
                                      512,
                                      Array [],
                                      "f070",
                                      "M320 400c-75.85 0-137.25-58.71-142.9-133.11L72.2 185.82c-13.79 17.3-26.48 35.59-36.72 55.59a32.35 32.35 0 0 0 0 29.19C89.71 376.41 197.07 448 320 448c26.91 0 52.87-4 77.89-10.46L346 397.39a144.13 144.13 0 0 1-26 2.61zm313.82 58.1l-110.55-85.44a331.25 331.25 0 0 0 81.25-102.07 32.35 32.35 0 0 0 0-29.19C550.29 135.59 442.93 64 320 64a308.15 308.15 0 0 0-147.32 37.7L45.46 3.37A16 16 0 0 0 23 6.18L3.37 31.45A16 16 0 0 0 6.18 53.9l588.36 454.73a16 16 0 0 0 22.46-2.81l19.64-25.27a16 16 0 0 0-2.82-22.45zm-183.72-142l-39.3-30.38A94.75 94.75 0 0 0 416 256a94.76 94.76 0 0 0-121.31-92.21A47.65 47.65 0 0 1 304 192a46.64 46.64 0 0 1-1.54 10l-73.61-56.89A142.31 142.31 0 0 1 320 112a143.92 143.92 0 0 1 144 144c0 21.63-5.29 41.79-13.9 60.11z",
                                    ],
                                    "iconName": "eye-slash",
                                    "prefix": "fas",
                                  }
                                }
                                size={15}
                                style={
                                  Object {
                                    "marginHorizontal": 5,
                                    "marginVertical": 12,
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                        <View
                          accessibilityRole="button"
                          accessibilityState={Object {}}
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {},
                              ]
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#007AFF",
                                    "fontSize": 18,
                                    "margin": 8,
                                    "textAlign": "center",
                                  },
                                ]
                              }
                            >
                              Login
                            </Text>
                          </View>
                        </View>
                      </View>
                      <Text
                        onPress={[Function]}
                        style={
                          Object {
                            "color": "#0000ff",
                            "marginTop": 20,
                            "textDecorationLine": "underline",
                          }
                        }
                      >
                        Continue as guest
                      </Text>
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          onPress={[Function]}
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "color": "#0000ff",
                              "margin": 8,
                              "textDecorationLine": "underline",
                            }
                          }
                        >
                          Create Account
                        </Text>
                        <Text
                          onPress={[Function]}
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "color": "#0000ff",
                              "margin": 8,
                              "textDecorationLine": "underline",
                            }
                          }
                        >
                          Forgot Password
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
